### PR TITLE
Offchain signing followup

### PR DESCRIPTION
**Pull request by rakanalh**
This is a follow up PR for #985 where some feedback was provided after the PR was merged.

### CODE REVIEW

File: runtime/common/src/parachains.rs [hunk 1]
```diff
@@ -813,12 +806,20 @@ impl<T: Trait> Module<T> {
...
 			);
+		// Make sure at least one account has successfully
+		// submitted the transaction.
+		let failed = results.iter().filter(|(_, res)| {
+			res.is_err()
+		});
+		if failed.count() == results.len() {
+			return None
+		}
 		return Some(())
```
Comment (bkchr) [inline runtime/common/src/parachains.rs:823]:
```suggestion
		Signer::<T, T::AuthorityId>::all_accounts()
			.send_signed_transaction(
				move |_account| {
					Call::report_double_vote(report.clone())
				}
			)
			.iter()
			.find_map(|res| res.ok().map(|_| ()))
```

File: runtime/common/src/parachains.rs [hunk 2]
```diff
@@ -64,23 +68,12 @@ use system::{
...
-	impl system::offchain::AppCrypto<<Sr25519Signature as Verify>::Signer, Sr25519Signature> for AuthorityId {
-		type RuntimeAppPublic = Public;
-		type GenericSignature = Sr25519Signature;
-		type GenericPublic = sp_core::sr25519::Public;
-	}
+// An `AppCrypto` type to facilitate submitting signed transactions.
+pub struct FishermanAuthorityId;
+impl system::offchain::AppCrypto<<Signature as Verify>::Signer, Signature> for FishermanAuthorityId {
+	type RuntimeAppPublic = FishermanId;
+	type GenericSignature = sr25519::Signature;
```
Comment (bkchr) [inline runtime/common/src/parachains.rs:75]: This is really over-complicated code...

Comment (tomusdrw) [reply runtime/common/src/parachains.rs:75]: Noted here: https://github.com/paritytech/substrate/issues/5662 Unfortunately using existing traits (`IsWrappedBy`) is non trivial and it is still to be addressed in substrate. But yeah, this code is definitely redundant.

Comment (bkchr) [reply runtime/common/src/parachains.rs:75]: https://github.com/paritytech/substrate/blob/999f4832e715f20a17fc2a3060bbe2ec88e7dbff/primitives/application-crypto/src/lib.rs#L204 this is even much easier.

**Comment by tomusdrw** [comment]
>https://github.com/paritytech/substrate/blob/999f4832e715f20a17fc2a3060bbe2ec88e7dbff/primitives/application-crypto/src/lib.rs#L204 this is even much easier.

But this is `AppKey`, which I believe is not supposed to be used in the runtime. If it is then we should remove `RuntimeAppPublic` cause now the only thing we have is confusion.

**Comment by rphmeier** [comment]
Sorry to review after merge again, but I still prefer to have this key fully defined in the primitives crate. I don't know what it's doing in runtime/parachains.

**Pull request merged**